config.go

```diff
@@ -0,0 +1,231 @@
+	}
+	if q.SQL == "" {
+		return fmt.Errorf("SQL statement required for query [%s]", q.Name)
+	}
+	if q.Timeout == 0 {
+		return fmt.Errorf("Timeout must be greater than zero for query [%s]", q.Name)
+	}
+	if q.Interval == 0 {
+		return fmt.Errorf("Interval must be greater than zero for query [%s]", q.Name)
+	}
+	return nil
+}
+
+func loadConfig(file string) (*Config, error) {
+	log.Printf("Load config from file [%s]", file)
+	b, err := ioutil.ReadFile(file)
+	if err != nil {
+		return nil, fmt.Errorf("Error reading config file: %s", err)
+	}
+	var c Config
+	if err := yaml.Unmarshal([]byte(b), &c); err != nil {
+		return nil, fmt.Errorf("Error decoding config file: %s", err)
+	}
+	c = appendDefaults(c)
+	if err := validateConfig(&c); err != nil {
+		return nil, err
+	}
+	return &c, err
+}
+
+func loadQueryConfig(queriesFile string, config *Config) (QueryList, error) {
+	log.Printf("Load queries from file [%s]", queriesFile)
+	// Read queries for request body.
+	file, err := os.Open(queriesFile)
+	if err != nil {
+		return nil, fmt.Errorf("Error opening queries file: %s", err)
+	}
+	defer file.Close()
+	return decodeQueries(file, config)
+}
+
+func useFallbackIfEmpty(value string, fallback string) string {
+	if value == "" {
+		return fallback
+	}
+	return value
+}
+
+func useFallbackIfZero(value time.Duration, fallback time.Duration) time.Duration {
+	if value == 0 {
+		return fallback
+	}
+	return value
+}
+
+func decodeQueries(r io.Reader, config *Config) (QueryList, error) {
+	if config == nil {
+		return nil, errors.New("Bug! Config must not be nil")
+	}
+
+	queries := make(QueryList, 0)
+	parsedQueries := make([]map[string]*Query, 0)
+
+	b, err := ioutil.ReadAll(r)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if err = yaml.Unmarshal(b, &parsedQueries); err != nil {
+		return nil, err
+	}
+
+	for _, data := range parsedQueries {
+		for k, q := range data {
+			q.Name = k
+			q.DataSourceRef = useFallbackIfEmpty(q.DataSourceRef, config.Defaults.DataSourceRef)
+			if q.Driver == "" {
+				if q.DataSourceRef != "" && len(config.DataSources) > 0 {
+					var ds = config.DataSources[q.DataSourceRef]
+					q.Driver = ds.Driver
+					q.Connection = ds.Properties
+				}
+			}
+			q.Interval = useFallbackIfZero(q.Interval, config.Defaults.QueryInterval)
+			q.Timeout = useFallbackIfZero(q.Timeout, config.Defaults.QueryTimeout)
+			if q.ValueOnError == "" && config.Defaults.QueryValueOnError != "" {
+				q.ValueOnError = config.Defaults.QueryValueOnError
+			}
+			q.DataField = strings.ToLower(q.DataField)
+			if err := validateQuery(q); err != nil {
+				return nil, err
+			}
+
+			queries = append(queries, q)
+		}
+	}
+	return queries, nil
+}
+
+func loadQueriesInDir(path string, config *Config) (QueryList, error) {
+	log.Printf("Load queries from directory [%s]", path)
+	queries := make(QueryList, 0)
+	files, err := ioutil.ReadDir(path)
+	if err != nil {
+		return nil, err
+	}
+	for _, f := range files {
```

Comment:
Thanks! It is good feedback to hear, will think about this next time. Also thinking if some of these general aesthetic "rules" shall be written in some kind of code guideline markdown file just for it to be more clear for me and others in the future.